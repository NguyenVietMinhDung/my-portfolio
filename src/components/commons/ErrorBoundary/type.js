export type Props = {
  children: any,
};

export type States = {
  hasError: boolean,
};
