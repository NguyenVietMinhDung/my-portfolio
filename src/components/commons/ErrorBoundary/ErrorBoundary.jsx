// @flow
// Use stateful component to catch thrown errors
// This will break Presentational and Container pattern
import React, { Component } from 'react';
import type { Props, States } from './type';

class ErrorBoundary extends Component<Props, States> {
  constructor(props: Props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError() {
    // Update state so the next render will show the fallback UI.
    return { hasError: true };
  }

  render() {
    const { hasError } = this.state;
    const { children } = this.props;

    if (hasError) {
      // You can render any custom fallback UI
      return (
        <h1 style={{ textAlign: 'center' }}>
          Opps! Something went wrong. Let&apos;s try again later.
        </h1>
      );
    }

    return children;
  }
}

export default ErrorBoundary;
