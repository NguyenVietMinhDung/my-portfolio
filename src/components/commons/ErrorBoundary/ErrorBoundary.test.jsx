import React from 'react';
import { shallow } from 'enzyme';
import ErrorBoundary from './ErrorBoundary';

const defaultProps = {
  children: <h1>Child Component</h1>,
};

describe('<ErrorBoundary />', () => {
  test('renders correctly', () => {
    const component = shallow(<ErrorBoundary {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
});
