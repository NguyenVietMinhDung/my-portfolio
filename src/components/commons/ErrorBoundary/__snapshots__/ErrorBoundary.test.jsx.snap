// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary /> renders correctly 1`] = `
<h1>
  Child Component
</h1>
`;
